### PR TITLE
[mlir][Vector] Remove trivial uses of vector.extractelement/vector.insertelement (1/N)

### PR DESCRIPTION
**Pull request by Groverkss**
This patch removes trivial usages of vector.extractelement/vector.insertelement. These operations can be fully represented by vector.extract/vector.insert. See https://discourse.llvm.org/t/rfc-psa-remove-vector-extractelement-and-vector-insertelement-ops-in-favor-of-vector-extract-and-vector-insert-ops/71116 for more information.

Further patches will remove more usages of these ops.

### CODE REVIEW

**Review by banach-space — APPROVED** [review]
Thanks, this is much easier to parse!

Vectorization.cpp, Linalg and Vector tests LGTM. I will let others review the remaining bits.

File: mlir/lib/Dialect/Vector/Transforms/VectorInsertExtractStridedSliceRewritePatterns.cpp
Comment (kuhar) [inline mlir/lib/Dialect/Vector/Transforms/VectorInsertExtractStridedSliceRewritePatterns.cpp:24]: After simplifying, do these helper make sense or should we just inline them?

**Review by kuhar — COMMENTED** [review]
Looks good, just one question/suggestion

File: mlir/lib/Dialect/Vector/Transforms/VectorInsertExtractStridedSliceRewritePatterns.cpp
Comment (Groverkss) [reply mlir/lib/Dialect/Vector/Transforms/VectorInsertExtractStridedSliceRewritePatterns.cpp]: inlined helpers

**Review by kuhar — APPROVED** [review]
LGTM, thanks for cleaning this up!

**Comment by dcaballe** [comment]
Hey, thanks for the cleanup! Have you found any performance differences after the replacement? As I noted somewhere else, one of the reasons this hasn't been done sooner is because `vector.extract/insert` and `vector.extractelement/vector.insertelement` were not on pair in terms of folders/canonicalizations/transformations. There really was a performance difference the last time I looked at this.

We should also leave a bit more time for reviewers before landing. Thanks!

**Comment by Groverkss** [comment]
> Hey, thanks for the cleanup! Have you found any performance differences after the replacement? As I noted somewhere else, one of the reasons this hasn't been done sooner is because `vector.extract/insert` and `vector.extractelement/vector.insertelement` were not on pair in terms of folders/canonicalizations/transformations. There really was a performance difference the last time I looked at this.
> 
> We should also leave a bit more time for reviewers before landing. Thanks!

Sorry, I need to be a bit better at waiting on the merge button.

My experience was that vector.extract/insert actually had better folders and more things were folding. But this is a handwavy analysis. Let me try to run some benchmarks with this patch in IREE. I will post them on the original thread (I just noticed you had the same comment there and I missed it, sorry for that). I'll make sure to also send the benchmark numbers with any further patches I send. Thanks for being cautious on this.

**Pull request merged**
